src/AppBundle/Twig/SourceCodeExtension.php

```diff
@@ -82,12 +86,15 @@ private function getControllerRelativePath()
 
         $className = get_class($this->controller[0]);
         $class = new \ReflectionClass($className);
+        $method = $class->getMethod($this->controller[1]);
 
+        $projectBaseDir = realpath($this->kernelRootDir.'/..');
         $absolutePath = $class->getFilename();
-        $pathParts = explode(DIRECTORY_SEPARATOR.'src'.DIRECTORY_SEPARATOR, $absolutePath);
-        $relativePath = 'src'.DIRECTORY_SEPARATOR.$pathParts[1];
+        $relativePath = ltrim(str_replace($projectBaseDir, '', $absolutePath), DIRECTORY_SEPARATOR);
+
+        $fileLink = strtr($this->fileLinkFormat, array('%f' => $absolutePath, '%l' => $method->getStartline()));
 
-        return $relativePath;
+        return empty($fileLink) ? $relativePath : new \Twig_Markup(sprintf('<a href="%s">%s</a>', $fileLink, $relativePath), 'UTF-8');
```

Comment:
you need to escape the link and the path. Otherwise, your Twig_Markup is not safe